pkg/reconciler/cluster/cluster.go

```diff
@@ -88,19 +88,27 @@ func (c *Controller) reconcile(ctx context.Context, cluster *v1alpha1.Cluster) e
 					!equality.Semantic.DeepEqual(pulledCrd.Labels, clusterCrd.Labels) {
 					pulledCrd.ResourceVersion = clusterCrd.ResourceVersion
 					_, err = c.crdClient.CustomResourceDefinitions().Update(logicalClusterContext, pulledCrd, v1.UpdateOptions{})
+					if errors.IsConflict(err) {
+						log.Printf("Conflict when applying CRD pulled from cluster %s for resource %s: %v ... Retrying\n", cluster.Name, resourceName, err)
+						c.enqueue(cluster)
```

Comment:
You're doing `c.enqueue` here, and the direct `c.Queue.AddRateLimited` in the deployment controller, [here](https://github.com/kcp-dev/kcp/pull/90/files#diff-82b04f39f7606c6c8a7d85e5217935522412f8c92dcc0e4736cfd679308ebf3dR101). Is there a reason to prefer one or the other?

Alternatively, could we just return a retryable error wrapper and have `process` take care of it for us? It feels like an abstraction violation to have this code know so much about requeueing.

Reply:
Fixed in commit https://github.com/kcp-dev/kcp/pull/90/commits/8f1b75d35617abd9faf5b3e99f1208d4109967c1